### DISCUSSION TITLE
Can a numeric mask be reactive?

### DISCUSSION THREAD

**Discussion by elcapo** [filamentphp/filament#6604] [Help]
I'm trying to make an amount field work with a dynamic mask that depends on the details (number of decimal places) of another currency field. Something in the lines of:

~~~php
Forms\Components\TextInput::make('decimals')
    ->reactive()
    ->default('1')
    ->required(),

Forms\Components\TextInput::make('amount')
    ->mask(fn (Mask $mask, \Closure $get) => $mask
        ->numeric()
        ->decimalPlaces((int) $get('decimals'))
        ->decimalSeparator('.')
    ),
~~~

## Expected Behavior

The behavior I'd expect is the second field to work by default with a mask of one decimal place, but automatically change to the number of decimal places indicated in the first field; whenever it changes.

## Current Behavior

The form works fine on page load for one decimal digit. But when **decimals** are changed, although the mask closure is executed, the **amount** field still works with one decimal place (no matter what was written in the first field). Only reloading the form with the new **decimals** value makes the mask react.

Am I doing something wrong? Is there a workaround for this?

**Comment by danharrin** [comment]
Currently, input masks cannot be dynamic. Sorry about that
